### PR TITLE
update poetry dependencies ❤️

### PR DESCRIPTION
**Pull request by plsholdmybeer**
Tests passed 🤷
👋😊
Not doing this often, had fun figuring things out, please do say if pull request is not valid or is missing something
Not that I did too much research on the topic, but this (as well as tiangolos sqlmodel) seem a bit underrated libraries. This kind of library (that "enforces" both async approach and unites concepts of models and schemas) along with a good logging middleware seem to me like a must for a fastapi project. I'm still studying though, I might be out of the loop and not looking at right places (although I'm not seeing any notices for better alternatives and tiangolos sqlmodel still has some activity). 

Another thing, this repo made me switch to poetry and learn its basics since pip really didn't care about a thing, installed the package and it did work, but when I saw the sqlalchemy deprecation logs and tried upgrading requirements, pip had an existential crisis downloading every single version of sqlalchemy (binary for windows), and kept trying for 15 minutes when I stopped it and just switched to poetry. Both approaches deserve ❤️ but I'm glad I found a reason to switch to a more elegant one. Thank you.

P.S @Kludex thank you for all the great work related to FastApi, your github profile is one of my main resources for R&D phase in my FastApi journey. FastApi is one of those technologies that "enforce" best practices of underlying technologies to do simple things and does it really well, making both itself and underlying technologies really shine in the process.

### CODE REVIEW

**Comment by Kludex** [comment]
> P.S @Kludex thank you for all the great work related to FastApi, your github profile is one of my main resources for R&D phase in my FastApi journey. FastApi is one of those technologies that "enforce" best practices of underlying technologies to do simple things and does it really well, making both itself and underlying technologies really shine in the process.

Thanks for the words! :heart: 

> please do say if pull request is not valid or is missing something

There are some code changes that you need to do based on https://github.com/Kludex/basesqlmodel/actions/runs/4569617636/jobs/8070926996?pr=7. :pray:

**Comment by plsholdmybeer** [comment]
had some issues with line endings but figured it out, hopefully the force push isn't a problem

**Comment by holdmybeer1min** [comment]
should I update the version?

**Comment by plsholdmybeer** [comment]
lint and test both pass now

File: basesqlmodel/main.py [hunk 1]
```diff
@@ -69,7 +69,7 @@ async def get(
         cls: Type[Self],
         session: AsyncSession,
         *args: BinaryExpression,
-        load_strategy: Dict[str, LoadStrategy] = None,
+        load_strategy: Union[Dict[str, LoadStrategy], None] = None,
```
Comment (Kludex) [inline basesqlmodel/main.py]:
```suggestion
        load_strategy: Dict[str, LoadStrategy] | None= None,
```

File: basesqlmodel/main.py [hunk 2]
```diff
@@ -82,7 +82,7 @@ async def get_multi(
...
```
Comment (Kludex) [inline basesqlmodel/main.py]:
```suggestion
        load_strategy: Dict[str, LoadStrategy] | None = None,
```

File: basesqlmodel/main.py [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 import inspect
 from functools import wraps
-from typing import Any, Callable, Dict, List, Literal, Type, TypeVar
+from typing import Any, Callable, Dict, List, Literal, Type, TypeVar, Union
```
Comment (Kludex) [inline basesqlmodel/main.py]:
```suggestion
from typing import Any, Callable, Dict, List, Literal, Type, TypeVar
```

File: pyproject.toml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "basesqlmodel"
-version = "0.1.0"
+version = "0.1.1"
```
Comment (Kludex) [inline pyproject.toml]:
```suggestion
version = "0.1.0"
```

File: tests/test_version.py [hunk 5]
```diff
@@ -2,4 +2,4 @@
 
 
 def test_version():
-    assert basesqlmodel.__version__ == "0.1.0"
+    assert basesqlmodel.__version__ == "0.1.1"
```
Comment (Kludex) [inline tests/test_version.py]:
```suggestion
    assert basesqlmodel.__version__ == "0.1.0"
```

**Review by Kludex — COMMENTED** [review]
I'll bump the version in another PR. :)

File: basesqlmodel/main.py [hunk 6]
```diff
@@ -69,7 +69,7 @@ async def get(
         cls: Type[Self],
         session: AsyncSession,
         *args: BinaryExpression,
-        load_strategy: Dict[str, LoadStrategy] = None,
+        load_strategy: Dict[str, LoadStrategy] | None= None,
```
Comment (Kludex) [inline basesqlmodel/main.py]:
```suggestion
        load_strategy: Dict[str, LoadStrategy] | None = None,
```

**Comment by Kludex** [comment]
Thanks @holdmybeer1min :)

File: basesqlmodel/main.py (hunk 1 above)
Comment (plsholdmybeer) [reply basesqlmodel/main.py]: > 

won't this fail on python versions < 3.10 ? That's the reason I used Union. I put the next lowest possible version in the pyproject.toml that will work with all existing dependencies and sqlmodel = "^0.0.8"

Comment (Kludex) [reply basesqlmodel/main.py]: The `__future__.annotations` make it pass.

Comment (plsholdmybeer) [reply basesqlmodel/main.py]: got it, thanks

**Pull request merged**